### ISSUE TITLE
Continuous translation threads for important discussions

### ISSUE THREAD

**Issue by zkat** [nodejs/inclusivity#102]
We currently have a mechanism for contributors/collaborators to request that a thread be watched for moderation to make sure things stay in line.

I would like to suggest developing a similar mechanism, but for translation: We could have various volunteers from the community who are willing to provide translations of issue/PR threads. They would then either translate particular posts for the requester, or simply mirror the entire thread in a separate issue, translating as the thread goes. They may also provide translation services for those members trying to participate with their own posts in such threads.

This is most likely not scalable to every single issue or discussion, but there are some particularly important threads that would certainly be relevant to the wider community, specially when they involve groups that have existing members who could use a bit of help.

This could also partly overlap in moderation in that we might be able to provide feedback to people responding to the thread on how to get to the point a bit sooner, instead of writing big long paragraphs.

tl;dr (just to dogfood my own idea):
- There are many non-native speakers in our community.
- Many of them are conversational in English, but may not be able to read complex technical blocks of text.
- I suggest having volunteers translate important threads when someone requests it.
- They would also help people respond in their native language (and translate it)
- The mechanism would be similar to flagging threads for moederation.

/cc @nodejs/localization @nodejs/intl @yosuke-furukawa @chrisdickinson

**Comment by zkat** [comment]
As per @ashleygwilliams, we may also want to ping someone from Mozilla about how they manage stuff like this (I'm kinda already on that, updates to come).

**Comment by srl295** [comment]
Would be a good use of GitHub Api and perhaps machine translation as a first cut. Thinking of something like translate wiki (how Wikipedia UI gets translated) with community editing and review. Interesting.

**Comment by TheAlphaNerd** [comment]
@srl295 does watson have a translation api?

**Comment by laosb** [comment]
Good to see that! So how can we address that?

> On Feb 9 2016, at 5:45 am, Kat Marchán &lt;notifications@github.com&gt;
wrote:  

>

> We currently have a mechanism for contributors/collaborators to request that
a thread be watched for moderation to make sure things stay in line.

>

> I would like to suggest developing a similar mechanism, but for translation:
We could have various volunteers from the community who are willing to provide
translations of issue/PR threads. They would then either translate particular
posts for the requester, or simply mirror the entire thread in a separate
issue, translating as the thread goes. They may also provide translation
services for those members trying to participate with their own posts in such
threads.

>

> This is most likely not scalable to every single issue or discussion, but
there are some particularly important threads that would certainly be relevant
to the wider community, specially when they involve groups that have existing
members who could use a bit of help.

>

> This could also partly overlap in moderation in that we might be able to
provide feedback to people responding to the thread on how to get to the point
a bit sooner, instead of writing big long paragraphs.

>

> tl;dr (just to dogfood my own idea):

>

>   * There are many non-native speakers in our community.

>   * Many of them are conversational in English, but may not be able to read
complex technical blocks of text.

>   * I suggest having volunteers translate important threads when someone
requests it.

>   * They would also help people respond in their native language (and
translate it)

>   * The mechanism would be similar to flagging threads for moederation.

>

> /cc
[@nodejs/localization](https://github.com/orgs/nodejs/teams/localization)
[@nodejs/intl](https://github.com/orgs/nodejs/teams/intl) [@yosuke-
furukawa](https://github.com/yosuke-furukawa)
[@chrisdickinson](https://github.com/chrisdickinson)

>

> —  
Reply to this email directly or [view it on GitHub](https://github.com/nodejs/
inclusivity/issues/102).![](https://github.com/notifications/beacon
/ACbWbfzI0mRekpw0j2OSqjtKQVcsInBWks5piQP-gaJpZM4HV43K.gif)

**Comment by srl295** [comment]
Indeed it does !

**Comment by timdp** [comment]
Great idea, but how would the UI look? Inline translations seem even more cumbersome than mirroring threads. Sounds more like something you'd want to accomplish using a specialized tool, which probably doesn't exist yet.

**Comment by mikeal** [comment]
I wonder if this will work in the "top down" way we're talking about it right now. In other words, will it work for us english speakers to say "this is important, people should translate it." 

My gut says that this may only work if we go about it in a bottom-up way, if a localization community sees a thread they find interesting and choose to continually translate it. Of course we could help this along by finding a good way to facilitate the notification and coordination, document best practices, etc.

**Comment by sergiolepore** [comment]
+1 to @mikeal's comment.

**Comment by zkat** [comment]
@mikeal note: I am not a native English speaker. I also suggested this as a reaction to the way @yosuke-furukawa has been doing tweet storms basically translating the gist of what some major npm and node-related threads have discussed. I think this is something that's already being done by a couple of folks, and we might just want to make resources available so that it's done in a more official/serious way as needed.

**Comment by mikeal** [comment]
@zkat awesome, whatever we can do to help, encourage or promote that kind of engagement the better :)

**Comment by zkat** [comment]
(I'm also hoping Yosuke and folks from the action localization community chime in on this thread at some point, since I also don't belong to any of those WGs)

**Comment by watilde** [comment]
This would be nice. I'm also one of them who have translated some updates to other language. The most hard thing I felt to continue translation work was to read all conversation to understand one topic, and I sometimes wanted a mechanism which can be requested a simple summary of a thread.

**Comment by cedced19** [comment]
This mechanism can be great. Actually in French translation group we haven't the time to translate all the news. If this works, we will save time. Also it means there is always someone to translate: that is not sure.

**Comment by yoannmoinet** [comment]
What would be possible, is assigning a tag to any issue / PR important enough to ask for translation.
Then, localization communities would simply reference the initial thread (to ping it) with their own issue / PR in a dedicated repo.

So it's not actually forcing anyone to do it, but only the more motivated communities present in that thread.

What do you think?

As a side note, I am in the French team too, and like @cedced19 said, engagement was really poor, and we've ended up being only 2-3 people translating / reviewing everything.

**Comment by mikeal** [comment]
One of the difficult things is notifying everyone. The easiest way to deal with that is to add team of the relevant people and @ reply them. Doing this to *every* localization team is a bit too much but we created a subteam(s) it could work.

**Comment by laosb** [comment]
Yes the engagement was really poor, in Chinese team too.

> On Feb 10 2016, at 10:28 pm, Yoann Moinet &lt;notifications@github.com&gt;
wrote:  

>

> What would be possible, is assigning a tag to any issue / PR important
enough to ask for translation.  
Then, localization communities would simply reference the initial thread (to
ping it) with their own issue / PR in a dedicated repo.

>

> So it's not actually forcing anyone to do it, but only the more motivated
communities present in that thread.

>

> What do you think?

>

> As a side note, I am in the French team too, and like
[@cedced19](https://github.com/cedced19) said, engagement was really poor, and
we've ended up being only 2-3 people translating / reviewing everything.

>

> —  
Reply to this email directly or [view it on GitHub](https://github.com/nodejs/
inclusivity/issues/102#issuecomment-182397154).![](https://github.com/notifica
tions/beacon/ACbWbQfM_5S4hul8eRwu_FLwl1DUIWkmks5pi0BjgaJpZM4HV43K.gif)

**Comment by carlosrymer** [comment]
@mikeal I agree. I'd add that it would also be useful to have a document listing these various teams and their members, a) so we know how to reference them and b) so we can keep track of teams that need help so people can pitch in.

**Comment by Fishrock123** [comment]
Would it be better to aim for writeups of possible options before a final decision is a made? Translating some of the "important" threads seems like an unreasonable time burden.

**Comment by kahwee** [comment]
I am having trouble understanding what is the problem that needs to be solved here. To me, the title proposes a solution which is "Continuous translation threads for important discussions".

Are we trying to:
a) Get non-English speakers involved in what is going to happen in Node.js
b) Get non-English speakers to participate in Node.js proposals in English via representatives who can translate back.
c) Get non-English speakers to participate in Node.js proposals in their native language

(a) this can be done as @Fiskrock123 suggests. Have a writeup, do machine translation. Post it to be a website like [Edge](https://dev.windows.com/en-us/microsoft-edge/platform/status/accessiblerichinternetapplicationswaiaria11) or [Webkit](https://webkit.org/status/)  and give it progress statuses.
(b) is slow
(c) is the hardest, we can't bridge language in a thread efficiently

I'm siding more with just doing (a) and only do (b) and (c) after that if there is significant demand. I realize I am clearly not promoting inclusivity here as much as I can but if we overdo this it may bog down contribution process.

**Comment by Charlotteis** [comment]
I believe the problem to be solved is that Node.js discussions are conducted in English (US or GB), which means a much larger population of the world cannot engage if they do not speak English. So, finding a way to get translations would mean that more people could even just read what was going on, which means that they are more likely to participate but this is not a requirement :)

Apologies if that's char-splaining, but your first statement was "what is the problem that needs to be solved here". I just wanted to make explicit that the goal doesn't have to be gaining participation, but rather allowing people to see what is going on so they feel less left out :)

**Comment by zkat** [comment]
@kahwee Basically what @Charlotteis said.

This originally came up because there were specific discussions (namely, the [discussion about converting APIs to Promises](https://github.com/nodejs/node/pull/5020)) where I wondered what non-native English speakers (or non-speakers!) would need to do to participate in a discussion that would, frankly, affect them greatly in the future as Node users.

So: How do we make sure that people who speak other languages are able to participate in this discussion *before* the active participants have reached a final conclusion? How do we get effective input from these folks?

**Comment by kahwee** [comment]
Thanks @zkat and @Charlotteis for the clarification! What do you think of sites like [ES Discuss](https://esdiscuss.org/1)

It's going to be really difficult to hold a discussion between English and non-English readers. Perhaps we can start by summarizing the proposals in Github and even encourage people to use simpler English words.

It will be nice if someone can translate for us but this may be too much to ask volunteers for.

**Comment by Trott** [comment]
This repo is inactive. If this is still a conversation worth having, the Community Committee might be a good place to take it (I think). https://github.com/nodejs/community-committee
